storage/flux/table.gen.go.tmpl

```diff
@@ -360,29 +703,211 @@ func (t *{{.name}}GroupTable) Do(f func(flux.ColReader) error) error {
 }
 
 func (t *{{.name}}GroupTable) advance() bool {
-RETRY:
-	a := t.cur.Next()
-	l := a.Len()
-	if l == 0 {
-		if t.advanceCursor() {
-			goto RETRY
+	if t.cur == nil {
+		// For group aggregates, we will try to get all the series and all table buffers within those series
+		// all at once and merge them into one row when this advance() function is first called.
+		// At the end of this process, t.advanceCursor() already returns false and t.cur becomes nil.
+		// But we still need to return true to indicate that there is data to be returned.
+		// The second time when we call this advance(), t.cur is already nil, so we directly return false.
+		return false
+	}
+	var arr *cursors.{{.Name}}Array
+	var len int
+	for {
+		arr = t.cur.Next()
+		len = arr.Len()
+		if len > 0 {
+			break
+		}
+		if !t.advanceCursor() {
+			return false
 		}
+	}
+
+	// handle the group without aggregate case
+	if t.gc.Aggregate() == nil {
+		// Retrieve the buffer for the data to avoid allocating
+		// additional slices. If the buffer is still being used
+		// because the references were retained, then we will
+		// allocate a new buffer.
+		colReader := t.allocateBuffer(len)
+		colReader.cols[timeColIdx] = arrow.NewInt(arr.Timestamps, t.alloc)
+		colReader.cols[valueColIdx] = t.toArrowBuffer(arr.Values)
+		t.appendTags(colReader)
+		t.appendBounds(colReader)
+		return true
+	}
 
+	aggregate, err := determine{{.Name}}AggregateMethod(t.gc.Aggregate().Type)
+	if err != nil {
+		t.err = err
 		return false
 	}
 
-	// Retrieve the buffer for the data to avoid allocating
-	// additional slices. If the buffer is still being used
-	// because the references were retained, then we will
-	// allocate a new buffer.
-	cr := t.allocateBuffer(l)
-	cr.cols[timeColIdx] = arrow.NewInt(a.Timestamps, t.alloc)
-	cr.cols[valueColIdx] = t.toArrowBuffer(a.Values)
-	t.appendTags(cr)
-	t.appendBounds(cr)
+	ts, v := aggregate(arr.Timestamps, arr.Values)
+	timestamps, values := []int64{ts}, []{{.Type}}{v}
+	for {
+		arr = t.cur.Next()
+		if arr.Len() > 0 {
+			ts, v := aggregate(arr.Timestamps, arr.Values)
+			timestamps = append(timestamps, ts)
+			values = append(values, v)
```

Comment:
Perhaps not relevant for this PR, but there is a bug here for the `ReadGroup(min/max)` case because the tag keys for the series (pre-grouping) are not preserved for selectors like min and max. @adrian-thurston is currently working on a fix for the 2.0 Cloud, and we'll need to ensure that fix also lands here once group agg pushdowns are enabled.

Reply:
I guess I'd like to be sure that we fix the `ReadGroup(min/max)` bug *before* we turn on pushdowns that expose the bug. How can we use our process to ensure this, any thoughts?